sql/parser2/eval.go

```diff
@@ -395,10 +435,18 @@ func EvalExpr(expr Expr, env Env) (Datum, error) {
 		return null, fmt.Errorf("column \"%s\" not found", t)
 
 	case Tuple:
-		if len(t) != 1 {
-			return null, fmt.Errorf("unsupported expression type: %T: %s", expr, expr)
+		if len(t) == 1 {
+			return EvalExpr(t[0], env)
```

Comment:
Can this lead to unexpected behavior (something like `true == [true]`)? Why isn't this better off being treated as below?